### PR TITLE
Add design doc to kubevela and update readme

### PR DESCRIPTION
**Pull request by resouer**
As the title said.

### CODE REVIEW

**Review by wonderflow — APPROVED** [review]

File: DESIGN.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+#### Solution
+
+We decide to build KubeVela core with [OAM Kubernetes runtime](https://github.com/crossplane/oam-kubernetes-runtime) which already implemented the building block features such as standalone workload type and trait controllers. Whether a given trait could work with certain workload types is also clarified as `appliesTo` field of trait definition. For platform builders, OAM Kubernetes Runtime also provided a [lightweight library](https://github.com/crossplane/oam-kubernetes-runtime/blob/2be3900a3817aed570de9ec353e6ab0b50e100f0/pkg/controller/v1alpha2/core/traits/manualscalertrait/manualscalertrait_controller.go#L42) to implement trait controller so it doesn't need to care about workload kinds.
+
+##### Capability Register and Discovery
+
+KubeVela leverages [OAM definition objects](https://github.com/oam-dev/spec/blob/master/4.workload_definitions.md) to register and discover workloads and traits:
```
Comment (zzxwill) [inline DESIGN.md:80]: Should we also link trait definition objects here https://github.com/oam-dev/spec/blob/master/6.traits.md#trait-definition?

File: DESIGN.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+```
+
+However, Kubernetes API should not be the end game. Declarative YAML objects are great to build platforms like KubeVela with but when directly exposed to end users, it creates heavy mental burden and high learning curve. For example, even with highest level abstraction, the default [containerized workload schema](https://github.com/oam-dev/spec/blob/master/core/workloads/schema/containerized_workload.md) of OAM still has dozens fields to learn with many rough edges to be aware.
+
+Thus KubeVela intends to introduce a lightweight user facing layer on top of its Kubernetes API objects with following goals in mind:
+
+- Shorten the learning curve of new developers. Most capabilities in KubeVela are developed by big
+companies that run very complex workloads. However, for the bigger developer community, the new user facing layer will provide a much simpler path to on-board KubeVela.
+- Developers can describe their applications and behavior of their components without making assumptions on availability of specific Kubernetes API. For instance, a developer will be able to model auto-scaling needs without referring to the CRD of auto-scaling trait.
```
Comment (zzxwill) [inline DESIGN.md:111]: +1

File: DESIGN.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+However, we don't assume users need to learn or write CUElang directly. Instead, the reason we choose CUElang is it's just perfect to help us create developer facing tools based on it. Thus, KubeVela introduced three most common tools based to CUElang for developers to use:
+
+1. A command line tool.
+2. A GUI dashboard.
+3. A Docker Compose style `appfile`.
+
+An example for `vela cli`:
+
+```console
+$ vela comp deploy frontend -t webservice --image oamdev/testapp:v1 --port 80 --app helloworld
```
Comment (zzxwill) [inline DESIGN.md:128]:
```suggestion
$ vela comp run frontend -t webservice --image oamdev/testapp:v1 --port 80 --app helloworld
```
Currently we use `vela comp run` to create an component. `deploy` also looks nice.

**Review by zzxwill — APPROVED** [review]
/lgtm

File: DESIGN.md (hunk 3 above)
Comment (wonderflow) [reply DESIGN.md:128]: I have already refactored to deploy yesteday

**Pull request merged**